<!doctype html>
<html lang="en">

	<head>
		<meta charset="utf-8">

		<title>reveal.js - The HTML Presentation Framework</title>

		<meta name="description" content="A framework for easily creating beautiful presentations using HTML">
		<meta name="author" content="Hakim El Hattab">

		<meta name="apple-mobile-web-app-capable" content="yes" />
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />

		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<link rel="stylesheet" href="css/reveal.min.css">
		<link rel="stylesheet" href="css/theme/default.css" id="theme">

		<!-- For syntax highlighting -->
		<link rel="stylesheet" href="lib/css/zenburn.css">

		<!-- If the query includes 'print-pdf', include the PDF print sheet -->
		<script>
			if( window.location.search.match( /print-pdf/gi ) ) {
				var link = document.createElement( 'link' );
				link.rel = 'stylesheet';
				link.type = 'text/css';
				link.href = 'css/print/pdf.css';
				document.getElementsByTagName( 'head' )[0].appendChild( link );
			}
		</script>

		<!--[if lt IE 9]>
		<script src="lib/js/html5shiv.js"></script>
		<![endif]-->
	</head>

	<body>

		<div class="reveal">

			<!-- Any section element inside of this container is displayed as a slide -->
<div class="slides">
    <section class="stack" data-id="c6b684835016767de4d7e9285068488f">
        <section class="" data-id="1b45c339f726e6573dc2d27d82d5a462">
            <h1>Galaxy community conference 2014</h1>
<div>Johns Hopkins, Baltimore, USA</div>
            <div>
                <br>
            </div>
            <div>conference report by&nbsp;</div>
            <div>Peter van Heusden &lt;pvh@sanbi.ac.za&gt;</div>
        </section>
        <section class="" data-id="369d39a61877b3dcd7c291850af1aa0a">
            <h2>GCC2014: social Media ? YES!</h2>
<div>
            <ul>
<li>#GCC2014 <a href="https://wiki.galaxyproject.org/Events/GCC2014">website</a>&nbsp;links to blogs and Storify of twitter stream</li>
<li>Online spaces key to Galaxy 'management':</li>
<ul>
<li style="padding-top: 0.25em; padding-bottom: 0.25em; vertical-align: middle;">IRC: #galaxyproject on irc.freenode.net</li>
<li style="padding-top: 0.25em; padding-bottom: 0.25em; vertical-align: middle;">Trello boards for <a href="https://trello.com/b/75c1kASa/galaxy-development">Galaxy development</a>&nbsp;and <a href="https://trello.com/b/3YgU637b/gcc2014-hackathon">hackathon</a>
</li>
</ul>
<li>Galaxy user support: Galaxy <a href="https://biostar.usegalaxy.org/">Biostar</a>
</li>
<li>Last remaining email holdout:</li>
<ul>
            <li>Galaxy-Dev mailing list (for how long?)</li>
        </ul>
</ul>
    </div>
</section>
</section>
<section class="stack" data-id="53455470f9abe6921c7fc7a663af5cf9">
    <section class="" data-id="54bdadd18db5ca12f056a31c5569de42">
        <h2>GCC2014: Hackathon</h2>
<div>
<ul>
<li>
<span>40 people</span>
        <br>
</li>
<li>Large pile of snacks, coffee and drinks</li>
<li>Two (+) days</li>
<li>
<a href="https://wiki.galaxyproject.org/Documents/Presentations/GCC2014?action=AttachFile&amp;do=view&amp;target=Hackathon.pdf">Outcomes</a>:</li>
<ul>
<li>
Most active 'idea' cards on Trello board:
</li>
<ol>
<li><a href="https://www.docker.com/">Docker:</a> wrapping Galaxy tools or Galaxy itself in Docker container</li>
<li>API: <a href="http://bioblend.readthedocs.org/en/latest/">BioBlend</a> Python wrapper around API</li>
</ol>
<li><strong>Great</strong> interaction with fellow developers (core/non-core)</li>
</ul>

<li>Architecture hacking:</li>
<ul>
<li>Sub-workflows</li>
<li>Collections as Galaxy tool inputs (and outputs)</li>
<li>Thoughts on federated Galaxy (GalaxyFarm)</li>
</ul>
</ul>
</div>
</section>
<section class="" data-id="5db1ca445386e66eb35bb3c2568b7d25">
    <h2>GCC2014: GALAXY ARCHITECTURE IDEAS</h2>
<div>
    <ul>
<li>Current Galaxy architecture still drawing from the model that there is a user clicking 'submit' on a job</li>
<li>At scale, this isn't a good model:</li>
<ul>
<li>Workflow runner: workflows are run as jobs, therefore fastest turnaround time is the time to execute a job (on typically cluster, at least 30 seconds)</li>
<li>Galaxy front-end / back-end splitting</li>
<ul>
<li>Front-end increasingly written in Javascript with <a href="http://backbonejs.org/">Backone</a>, <a href="http://underscorejs.org/">Underscore</a> and <a href="http://handlebarsjs.com/">Handlebars</a>: more responsive UI</li>
<li>Pulsar (used to be LWR) can run on compute resource, no shared filesystem, communicate with front-end using RabbitMQ (used for Galaxy Main)</li>
</ul>
</ul>
</ul>
</div>
</section>
<section class="" data-id="24f1a7b310e7ce900c2dc9f7e06929ff">
    <h2>BCBIO-NEXTGEN ARCHITECTURE</h2>
    <div>
        <ul>
            <li>NGS analysis automated (but not by Galaxy)</li>
            <li>Disk usage scaling:</li>
            <ul>
                <li>Pipeline analysis steps together saves disk I/O</li>
                <li>Avoid storing intermediate data: 6x size of final data</li>
                <li>bcbio-nextgen scaling driven by need to process 1500 WGS&nbsp;</li>
            </ul>
            <li>Parallelisation:</li>
            <ul>
                <li>Split variant calling by callable region</li>
                <li>"Smart" parallelism driven by knowledge of underlying data</li>
            </ul>
            <li>Configurability is the enemy of scaling</li>
            <ul>
                <li>Supporting lots of scenarios makes optimisation hard</li>
            </ul>
            <li>bcbio-nextgen now available as Docker image and runnable as Galaxy tool</li>
        </ul>
    </div>
</section>
</section>
<section class="stack">
    <section class="" data-id="e7cee0b1767d00d623a36eed2c6afd33">
    <h2>STATE Of THE GALAXY<font style="font-size: 24px;">
        <a href="https://wiki.galaxyproject.org/Documents/Presentations/GCC2014?action=AttachFile&amp;do=view&amp;target=NekrutenkoTaylor.pdf">1</a>
    </font>
</h2>
<div>
    <ul>
        <li>Developments in the toolshed</li>
        <li>Investigations in new ways of job running</li>
        <li>&nbsp;Galaxy/biostar: Galaxy Q&amp;A</li>
        <li>New work on visualisations</li>
        <li>Data managers: tools for handling reference data</li>
        <li>Data collections (and parameter sweeps)</li>
    </ul>
</div>
</section>
<section class="" data-id="dcbb13fdd6ab266896b24b78ccb33116">
    <h2>GALAXY TOOLSHED, 2014 EDITION</h2>
    <div>
        <ul>
            <li>Toolshed is replacing hand-editing of tool_conf.xml</li>
            <li>
                <span>Recommended to run local toolshed (SANBI does)</span>
            </li>
            <ul>
                <li>
                    <span>Tool development happens in Mercurial repos</span>
                </li>
                <li>
                    <span>Galaxy server tool updates don't require restarts</span>
                </li>
            </ul>
            <li>
                <span>Tools can have dependencies:</span>
            </li>
            <ul>
                <li>
                    <span>On other tool repositories (other tools or data types)</span>
                </li>
                <li>
                    <span>On packages (package manager for Galaxy ??)</span>
                </li>
                <ul>
                    <li>
                        <span>Huge role for Docker here in future</span>
                    </li>
                </ul>
            </ul>
            <li>
                <span>Galaxy tools can be moved between repositories as .tar.gz</span>
                <br>
            </li>
        </ul>
    </div>
</section>
<section class="" data-id="f9c9dd5fbecc38b64e5c33d5de01833a">
    <h2>NEW WAYS OF JOB RUNNING</h2>
    <div>
        <ol>
            <li>
                <a href="https://github.com/galaxyproject/pulsar">Pulsar</a>/(ex-LWR)&nbsp;(<a href="https://github.com/jmchilton">John Chilton</a>)<span>: Job-execution environment for when you don't have a shared filesystem</span>
            </li>
            <ul>
                <li>
                    <span>Appears to Galaxy as just another job runner</span>
                    <br>
                </li>
                <li>
                    <span>Future direction: workflow engine needed!</span>
                </li>
            </ul>
            <li>Galaxy on a resource-limited Cluster (<a href="http://www.usit.uio.no/om/organisasjon/uav/itf/ft/ansatte/nikolaiv/index.html">Nikolay Vazov</a>, UIO):<br>
            </li>
            <ul>
                <li>
                    <a href="https://wiki.galaxyproject.org/Events/GCC2014/TrainingDay?action=AttachFile&amp;do=view&amp;target=UserProjectMgmt.pdf">Integration</a> with <a href="http://www.clusterresources.com/products/gold-allocation-manager.php">Gold Allocation Manager</a>
                </li>
                <li>Essentially fork of Galaxy, required substantial modification to the code base</li>
                <li>Exposes resource limits to Galaxy users, integrates with resource accounting</li>
            </ul>
        </ol>
    </div>
</section>
<section class="" data-id="c0b506755e39ceaf7c90c41b20db791d">
    <h2>Galaxy &amp; the users</h2>
    <div>
        <ul>
            <li>Galaxy-User mailing list discontinued</li>
            <li>Replaced by Galaxy Biostar site</li>
            <ul>
                <li>Allows easier searching for commonly answered questions</li>
                <li>Support still appears to be driven by Galaxy Team</li>
            </ul>
            <li>Galaxy usability:&nbsp;</li>
            <ul>
                <li>Trello cards open calling for extensions to track user interaction with Galaxy</li>
                <li>Difficulty to work seriously on topic without detailed Human Computer Interaction studies</li>
            </ul>
            <li>Galaxy User BoF:&nbsp;</li>
            <ul>
                <li>What is a User? Given the large number of technical users, technical questions tend to dominate "how do I?" space</li>
            </ul>
        </ul>
    </div>
</section>
<section class="" data-id="21a4a86ee67697ccf9a27526db323d54">
    <h2>VISUALISATION</h2>
    <div>
        <ul>
            <li>
                <a href="https://wiki.galaxyproject.org/Learn/Visualization/Charts">Galaxy Charts</a>&nbsp;allows in-browser visualisation of tabular data</li>
                <ul>
                    <li>supports area, bar, box, line, pie charts and scatter plots</li>
                </ul>
                <li>Work ongoing to support visualisations installable as tools</li>
                <li>
                    <a href="http://www.biomedcentral.com/1471-2164/14/397">Not only Trackster</a>, now also:</li>
                    <ul>
                        <li>PhyloViz: phylogeny tree viewer</li>
                        <li>Cirster: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2752132/?tool=pubmed">circos</a> visualisations</li>
                        <li>Sweepster: integrates parameter sweep exploration of genomic regions with Trackster</li>
                    </ul>
                    <li>Documentation is still patchy</li>
                </ul>
            </div>
        </section>
        <section class="" data-id="63a14b7d43af43de9abccd509e2c5691">
            <h2>DATA MANAGERS</h2>
            <div>
                <ul>
                    <li>Galaxy install has tool-data/ directory that configures shared tool data</li>
                    <ul>
                        <li>e.g. available genome builds</li>
                    </ul>
                    <li>
                        <a href="https://wiki.galaxyproject.org/Admin/Tools/DataManagers">Data managers</a> allow configuring this from the web interface<br>
                        </li>
                        <ul>
                            <li>
                                <span>In the Admin panel (for site-admins only)</span>
                            </li>
                            <li>
                                <span>Run in similar way to normal tools, but output of run isn't in History, but rather visible via Admin panel</span>
                            </li>
                            <li>
                                <span>Can be installed from a toolshed</span>
                            </li>
                        </ul>
                    </ul>
                </div>
            </section>
            <section class="" data-id="fd50367e0784f924c6c2fa5d5e07f64a">
                <h2>DATASET COLLECTIONS</h2>
                <div>
                    <ul>
                        <li>Galaxy core now supports <a href="https://wiki.galaxyproject.org/Documents/Presentations/GCC2014?action=AttachFile&amp;do=view&amp;target=Chilton.pdf">dataset collection</a> types:</li>
                        <ul>
                            <li>List</li>
                            <li>Pair (forward / reverse)</li>
                        </ul>
                        <li>Allow operation on a collection:</li>
                        <ul>
                            <li>List can be used as tool input</li>
                        </ul>
                        <li>Tools that need paired reads can take single pair param</li>
                        <li>Support in the code (and via API) is currently further along that support in the UI</li>
                        <li>Much work needed</li>
                        <ul>
                            <li>Filtering on collections, collections as tool outputs, workflow re-run / resume</li>
                            <li>Parameter sets (like Sweepster, but more general)</li>
                        </ul>
                    </ul>
                </div>
            </section>
    </section>
            <section class="stack" data-id="702bb379b678f5db632e38ef318164e4">
                <section class="" data-id="adfce7f73d4dc135438a6741215c579e">
                    <h2>CONCLUSION</h2>
                    <div>
                        <ul>
                            <li>Large-scale Galaxy is clearly on the roadmap for the future</li>
                            <ul>
                                <li>New workflow scheduler and execution engine</li>
                                <li>Continue and extend work on collection types</li>
                                <li>Develop UI elements for working at scale</li>
                            </ul>
                            <li>Galaxy can do many quite different things</li>
                            <ul>
                                <li>Sometimes by forking Galaxy code (but then you lose community)</li>
                                <li>Not all code paths are equal: frequently used code paths get bugs noticed and fixed earlier</li>
                            </ul>
                            <li>Non-Galaxy approaches continue to proliferate: <a href="https://ccg.murdoch.edu.au/yabi/login/?next=/yabi/">YABI</a>, <a href="https://bcbio-nextgen.readthedocs.org/en/latest/">bcbio-nextgen</a>, <a href="https://arvados.org/">Arvados</a>, <a href="http://bdgenomics.org/">ADAM</a>
                        </li>
                        <ul>
                            <li>Lots to learn from</li>
                        </ul>
                    </ul>
                </div>
            </section>
            <section class="" data-id="e839843dbfa895ec02508f1f280ced9c">
                <h2><span style="vertical-align: middle; display: inline-block;">SANBI GALAXY PLANS</span><img src="images/SANBIlogo.png" style="background-color: transparent; vertical-align: middle; border: none; margin: 0px;  height: 75px; max-height: none; max-width: none;"></h2>
                <div>
                    <ul>
                        <li>SANBI committed to working on making Galaxy more capable of expressing the workflows we already use</li>
                        <li>Combination of bug fixing and internals enhancement</li>
                        <li>One post-doc, one developer (hopefully two soon)</li>
                        <li>Key goals align with some of Galaxy community</li>
                        <ul>
                            <li>Workflows as first-class objects</li>
                            <li>Data collections everywhere</li>
                            <li>Workflows scheduled as workflows</li>
                            <li>Fine grained execution tracking</li>
                            <li>Data handling improvements (is piping possible?)</li>
                            <li>New on-disk storage types (<a href="http://avro.apache.org/">Apache Avro</a>?)</li>
                        </ul>
                        <li>Continue collaboration with <a href="https://pipelines.sanbi.ac.za/index.php/Main_Page">Pipelines</a> group</li>
                    </ul>
                </div>
            </section>
        </section>
    </div>
		</div>

		<script src="lib/js/head.min.js"></script>
		<script src="js/reveal.min.js"></script>

		<script>

			// Full list of configuration options available here:
			// https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				controls: true,
				progress: true,
				history: true,
				center: true,

				theme: Reveal.getQueryHash().theme, // available themes are in /css/theme
				transition: Reveal.getQueryHash().transition || 'default', // default/cube/page/concave/zoom/linear/fade/none

				// Parallax scrolling
				// parallaxBackgroundImage: 'https://s3.amazonaws.com/hakim-static/reveal-js/reveal-parallax-1.jpg',
				// parallaxBackgroundSize: '2100px 900px',

				// Optional libraries used to extend on reveal.js
				dependencies: [
					{ src: 'lib/js/classList.js', condition: function() { return !document.body.classList; } },
					{ src: 'plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
					{ src: 'plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
					{ src: 'plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } }
				]
			});

		</script>

	</body>
</html>
